packages/apollo-gateway/src/index.ts

```diff
@@ -78,32 +98,33 @@ export class ApolloGateway implements GraphQLService {
     loglevelDebug(this.logger);
 
     // And also support the `debug` option, if it's truthy.
-    if (config.debug === true) {
+    if (this.config.debug === true) {
       this.logger.enableAll();
     }
 
-    if (isLocalConfig(config)) {
-      this.createSchema(config.localServiceList);
+    if (isLocalConfig(this.config)) {
+      this.createSchema(this.config.localServiceList);
     }
 
     this.initializeQueryPlanStore();
   }
 
-  public async load() {
+  public async load(engineConfig?: EngineConfig) {
```

Comment:
Switched to `options?: { engine: GraphQLServiceEngineConfig }`. I think it makes sense in `load` though. Otherwise you need to start to enforce "Make sure you call `config(...)` before you call `load()`, but only if you are running in managed mode"-type APIs.